src/Microsoft.AspNet.OData.Shared/EdmChangedObjectCollection.cs

```diff
@@ -60,6 +70,317 @@ private void Initialize(IEdmEntityType entityType)
             _entityType = entityType;
             _edmType = new EdmDeltaCollectionType(new EdmEntityTypeReference(_entityType, isNullable: true));
             _edmTypeReference = new EdmCollectionTypeReference(_edmType);
+            
+        }
+
+        /// <summary>
+        /// Patch for Types without underlying CLR types
+        /// </summary>
+        /// <param name="originalCollection"></param>
+        /// <returns>ChangedObjectCollection response</returns>
+        internal EdmChangedObjectCollection Patch(ICollection<IEdmStructuredObject> originalCollection)
+        {
+            EdmODataAPIHandler apiHandler = new DefaultEdmODataAPIHandler(originalCollection, _entityType);
+
+            return CopyChangedValues(apiHandler);
+        }
+
+        /// <summary>
+        /// Patch for EdmChangedObjectCollection, a collection for IEdmChangedObject 
+        /// </summary>
+        /// <returns>ChangedObjectCollection response</returns>
+        internal EdmChangedObjectCollection Patch(EdmODataAPIHandler apiHandler, ODataEdmAPIHandlerFactory apiHandlerFactory)
```

Comment:
Most of the logic in this class is pretty similar to what we have in the DeltaSet<T> class.. Is there a way in which it can be consolidated. Esp the repeated logic.

Reply:
Wont fix now.
Refactor logic later.